pkg/internal/tests/cross-tests/adapt.go

```diff
@@ -17,13 +17,134 @@
 package crosstests
 
 import (
+	"context"
+	"fmt"
 	"math/big"
 
 	"github.com/hashicorp/terraform-plugin-go/tftypes"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+	"github.com/stretchr/testify/require"
 	"github.com/zclconf/go-cty/cty"
+
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/convert"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge/info"
+	shim "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/unstable/logging"
 )
 
+// InferPulumiValue is a special marker value that tells cross-tests to infer the Pulumi
+// value from the TF value.
+func InferPulumiValue() resource.PropertyMap {
+	return resource.PropertyMap{
+		inferPulumiValueKey: resource.NewProperty(&inferPulumiValueSecret),
+	}
+}
+
+const inferPulumiValueKey = "inferPulumiValue"
+
+var inferPulumiValueSecret = resource.Secret{}
+
+func shouldInferredPulumiValue(m resource.PropertyMap) bool {
+	v, ok := m[inferPulumiValueKey]
+	return ok && v.IsSecret() && v.SecretValue() == &inferPulumiValueSecret
+}
+
+func inferPulumiValue(t T, schema shim.SchemaMap, infos map[string]*info.Schema, v cty.Value) resource.PropertyMap {
```

Comment:
I find it a bit confusing that this function and https://github.com/pulumi/pulumi-terraform-bridge/pull/2505/files#diff-923fa63a12c441bd1b41d16548d1eba8f7a3ab0c628eed9955c512a907bf713eR38 have the same name but they are quite different. Should we rename one?

Reply:
I added a comment to clarify.